taxi/views.py

```diff
@@ -79,6 +81,24 @@ class CarDeleteView(LoginRequiredMixin, generic.DeleteView):
     success_url = reverse_lazy("taxi:car-list")
 
 
+@login_required
+def assign_me_to_car(request, pk):
+    car = get_object_or_404(Car, pk=pk)
+    if request.user not in car.drivers.all():
+        car.drivers.add(request.user)
+
+    return redirect("taxi:car-detail", pk=car.pk)
+
+
+@login_required
+def delete_me_from_car(request, pk):
+    car = get_object_or_404(Car, pk=pk)
+    if request.user in car.drivers.all():
+        car.drivers.remove(request.user)
+
+    return redirect("taxi:car-detail", pk=car.pk)
```

Comment:
use .exists() so that db uses WHERE and EXISTS at the same time because otherwise you firstly fetch 1 000 000 records and then python loops through all records to check "in" condition